docs/upgrading.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading the Agent
+
+This document provides user guidance around agent upgrades and versioning.
+Like every other piece of software included in your project, the agent 
+should be upgraded frequently enough to pick up ongoing enhancements 
+while allowing room for testing. 
+
+## Watching For New Releases
+
+How can you find out about new releases? The best way is to "Watch" this repo
+from the "Watch" button in the upper-right corner of the main repository 
+page. If you don't want to be notified for every pull request and issue, you can 
+choose "Custom" under "Watch" and select just the "Releases" checkbox.
+
+Another approach is to just check in regularly with the GitHub 
+[releases page](https://github.com/signalfx/splunk-otel-java/releases) for this 
+project. 
+
+## When Does Splunk Release?
+
+Because we are a downstream distribution of [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation),
+we generally release within a few working days of every upstream release. We also release when we 
+have new features or enhancements that we want to make available.
+
+We contribute to upstream OpenTelemetry and are typically aware of the release cycle
```

Comment:
"typically" is also vague. Are you not aware of the release cycle? 

What is "upstream OpenTelemetry"? I assume that "over there" refers to something related to upstream OpenTelemetry.

Reply:
> "typically" is also vague. Are you not aware of the release cycle?

We usually are. There could be circumstances where a release happens quickly or without our direct involvement (due to scheduling/priorities, etc).
 
> What is "upstream OpenTelemetry"? I assume that "over there" refers to something related to upstream OpenTelemetry.

I think upstream/downstream descriptors are pretty idiomatic in the open source world. "Upstream OpenTelemetry" here means https://github.com/open-telemetry/opentelemetry-java-instrumentation (the community instrumentation project on which our distribution is based).